// pui-react-flex-grids/flex-grids.js
/*(c) Copyright 2015 Pivotal Software, Inc. All Rights Reserved.*/
'use strict';

exports.__esModule = true;
exports.FlexCol = exports.Grid = undefined;

var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');

var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _puiReactHelpers = require('pui-react-helpers');

var _classnames6 = require('classnames');

var _classnames7 = _interopRequireDefault(_classnames6);

require('pui-css-flex-grids');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Grid = exports.Grid = function (_React$PureComponent) {
  (0, _inherits3.default)(Grid, _React$PureComponent);

  function Grid() {
    (0, _classCallCheck3.default)(this, Grid);
    return (0, _possibleConstructorReturn3.default)(this, _React$PureComponent.apply(this, arguments));
  }

  Grid.prototype.render = function render() {
    var _props = this.props,
        gutter = _props.gutter,
        props = (0, _objectWithoutProperties3.default)(_props, ['gutter']);

    var newProps = (0, _puiReactHelpers.mergeProps)(props, { className: (0, _classnames7.default)('grid', gutter ? '' : 'grid-nogutter') });
    return _react2.default.createElement('div', newProps);
  };

  return Grid;
}(_react2.default.PureComponent);

Grid.propTypes = {
  gutter: _propTypes2.default.bool
};
Grid.defaultProps = {
  gutter: true
};
var FlexCol = function FlexCol(props) {
  var _classnames, _classnames2, _classnames3, _classnames4, _classnames5;

  var col = props.col,
      fixed = props.fixed,
      grow = props.grow,
      alignment = props.alignment,
      contentAlignment = props.contentAlignment,
      breakpoint = props.breakpoint,
      other = (0, _objectWithoutProperties3.default)(props, ['col', 'fixed', 'grow', 'alignment', 'contentAlignment', 'breakpoint']);


  var colClassName = (0, _classnames7.default)((_classnames = {}, _classnames['col-' + col] = col, _classnames));

  var fixedClassName = (0, _classnames7.default)({
    'col-fixed': fixed
  });

  var growClassName = (0, _classnames7.default)((_classnames2 = {}, _classnames2['col-grow-' + grow] = grow, _classnames2));

  var alignmentClassName = (0, _classnames7.default)((_classnames3 = {}, _classnames3['col-align-' + alignment] = alignment, _classnames3));

  var contentAlignmentClassName = (0, _classnames7.default)((_classnames4 = {}, _classnames4['col-' + contentAlignment] = contentAlignment, _classnames4));

  var breakpointClassName = (0, _classnames7.default)((_classnames5 = {}, _classnames5['col-' + breakpoint] = breakpoint, _classnames5));

  var className = (0, _classnames7.default)('col', colClassName, fixedClassName, growClassName, alignmentClassName, contentAlignmentClassName, breakpointClassName);

  var newProps = (0, _puiReactHelpers.mergeProps)(other, { className: className });
  return _react2.default.createElement('div', newProps);
};

exports.FlexCol = FlexCol;
FlexCol.propTypes = {
  col: _propTypes2.default.number,
  fixed: _propTypes2.default.bool,
  grow: _propTypes2.default.number,
  alignment: _propTypes2.default.oneOf(['top', 'middle', 'bottom']),
  contentAlignment: _propTypes2.default.oneOf(['top', 'middle', 'bottom']),
  breakpoint: _propTypes2.default.oneOf(['sm', 'md', 'lg'])
};
