// pui-react-flex-grids/flex-grids.js
/*(c) Copyright 2015 Pivotal Software, Inc. All Rights Reserved.*/
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.FlexCol = exports.Grid = undefined;

var _defineProperty2 = require('babel-runtime/helpers/defineProperty');

var _defineProperty3 = _interopRequireDefault(_defineProperty2);

var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');

var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _puiReactHelpers = require('pui-react-helpers');

var _classnames6 = require('classnames');

var _classnames7 = _interopRequireDefault(_classnames6);

require('pui-css-flex-grids');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Grid = exports.Grid = function (_React$Component) {
  (0, _inherits3.default)(Grid, _React$Component);

  function Grid() {
    (0, _classCallCheck3.default)(this, Grid);
    return (0, _possibleConstructorReturn3.default)(this, (Grid.__proto__ || (0, _getPrototypeOf2.default)(Grid)).apply(this, arguments));
  }

  (0, _createClass3.default)(Grid, [{
    key: 'render',
    value: function render() {
      var _props = this.props,
          gutter = _props.gutter,
          props = (0, _objectWithoutProperties3.default)(_props, ['gutter']);

      var newProps = (0, _puiReactHelpers.mergeProps)(props, { className: (0, _classnames7.default)('grid', gutter ? '' : 'grid-nogutter') });
      return _react2.default.createElement('div', newProps);
    }
  }]);
  return Grid;
}(_react2.default.Component);

Grid.propTypes = {
  gutter: _propTypes2.default.bool
};
Grid.defaultProps = {
  gutter: true
};
var FlexCol = function FlexCol(props) {
  var col = props.col,
      fixed = props.fixed,
      grow = props.grow,
      alignment = props.alignment,
      contentAlignment = props.contentAlignment,
      breakpoint = props.breakpoint,
      other = (0, _objectWithoutProperties3.default)(props, ['col', 'fixed', 'grow', 'alignment', 'contentAlignment', 'breakpoint']);


  var colClassName = (0, _classnames7.default)((0, _defineProperty3.default)({}, 'col-' + col, col));

  var fixedClassName = (0, _classnames7.default)({
    'col-fixed': fixed
  });

  var growClassName = (0, _classnames7.default)((0, _defineProperty3.default)({}, 'col-grow-' + grow, grow));

  var alignmentClassName = (0, _classnames7.default)((0, _defineProperty3.default)({}, 'col-align-' + alignment, alignment));

  var contentAlignmentClassName = (0, _classnames7.default)((0, _defineProperty3.default)({}, 'col-' + contentAlignment, contentAlignment));

  var breakpointClassName = (0, _classnames7.default)((0, _defineProperty3.default)({}, 'col-' + breakpoint, breakpoint));

  var className = (0, _classnames7.default)('col', colClassName, fixedClassName, growClassName, alignmentClassName, contentAlignmentClassName, breakpointClassName);

  var newProps = (0, _puiReactHelpers.mergeProps)(other, { className: className });
  return _react2.default.createElement('div', newProps);
};

exports.FlexCol = FlexCol;
FlexCol.propTypes = {
  col: _propTypes2.default.number,
  fixed: _propTypes2.default.bool,
  grow: _propTypes2.default.number,
  alignment: _propTypes2.default.oneOf(['top', 'middle', 'bottom']),
  contentAlignment: _propTypes2.default.oneOf(['top', 'middle', 'bottom']),
  breakpoint: _propTypes2.default.oneOf(['sm', 'md', 'lg'])
};
